scipy/stats/_stats_py.py

```diff
@@ -5551,6 +5563,159 @@ def _mgc_stat(distx, disty):
+        If ``workers`` is an int the population is subdivided into ``workers``
+        sections and evaluated in parallel (uses ``multiprocessing.Pool
+        <multiprocessing>``). Supply ``-1`` to use all cores available to the
+        Process. Alternatively supply a map-like callable, such as
+        ``multiprocessing.Pool.map`` for evaluating the p-value in parallel.
+        This evaluation is carried out as ``workers(func, iterable)``.
+        Requires that `func` be pickleable. The default is ``1``.
+    is_twosamp : bool, optional
+        If `True`, a two sample test will be run. If ``x`` and ``y`` have
+        shapes ``(n, p)`` and ``(m, p)``, this optional will be overriden and
+        set to ``True``. Set to ``True`` if ``x`` and ``y`` both have shapes
+        ``(n, p)`` and a two sample test is desired. The default is ``False``.
+        Note that this will not run if inputs are distance matrices.
+    random_state : {None, int, `numpy.random.Generator`,
+                    `numpy.random.RandomState`}, optional
+
+        If `seed` is None (or `np.random`), the `numpy.random.RandomState`
+        singleton is used.
+        If `seed` is an int, a new ``RandomState`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``Generator`` or ``RandomState`` instance then
+        that instance is used.
+
+    Returns
+    -------
+    stat : float
+        The sample Dcorr test statistic within `[-1, 1]`.
+    pvalue : float
+        The p-value obtained via permutation.
+
+    
+    Notes
+    -----
+
+    #. Let x and y be (n,p) samples of random variables X and Y. Let D^x be the n×n 
+       distance matrix of x and D^y be the n×n be the distance matrix of y. 
+       The distance covariance is
+    
+    .. math::
+
+    Dcovnb(x,y) = (1/(n^2)) \ times trace(D^{y} H D^{y} H)
+
+    #. where tr(⋅) is the trace operator and H is defined as H=I−(1/n)J where I 
+       is the identity matrix and J is a matrix of ones. The normalized version 
+       of this covariance is distance correlation 1 and is
+    
+    .. math::
+
+    Dcorrnb(x,y) = {Dcovnb(x,y)}/{\sqrt{Dcovnb(x,x) Dcovnb(y,y)}}
+
+    #. This is a biased test statistic. An unbiased alternative also exists, 
+       and is defined using the following: 
+
+    #. Consider the centering process where 1(⋅) is the indicator function:
+    
+    .. math::
+
+    Cij^x = [\(Dsb^x - (1/(n-2)) \sum_{t=1}^{n} Dit^x - (1/(n-2)) \sum_{s=1}^{n} Dsj^x 
+             + (1/(n-2)(n+2)) \sum_{s,t=1}^{n} Dst^x]1\sb{i≠j}
+    
+    #. and similarly for Cy. 
+
+    #. Then, this unbiased Dcorr is:
+    
+    .. math:: 
+    
+    Dcovn(x,y)=(1n/(n−3))tr(CxCy)
+
+    #. The normalized version of this covariance 2 is:
+
+    .. math::
+
+    Dcorrn(x,y)={Dcovn(x,y)}/{\sqrt{Dcovn(x,x) Dcovn(y,y)}}
+
+    #. The p-value returned is calculated using a permutation test using 
+       hyppo.tools.perm_test. The fast version of the test uses hyppo.tools.chi2_approx.
+
```

Comment:
mention that Dcorr is equivalent to Energy, Hsic and cite nonpar manova via independence testing paper and exact equivalence of distance and kernel methods paper